ambuda/views/auth.py

```diff
@@ -105,14 +167,76 @@ def sign_out():
     return redirect(url_for(POST_AUTH_ROUTE))
 
 
-@bp.route("/recover", methods=["GET", "POST"])
-def recover():
-    form = RecoverForm()
+@bp.route("/reset-password", methods=["GET", "POST"])
+def get_reset_password_token():
+    """Email the user a password reset link."""
+    form = ResetPasswordForm()
     if form.validate_on_submit():
-        hashed = generate_password_hash(form.password.data)
-    else:
-        hashed = ""
-    return render_template("auth/recover.html", form=form, hashed=hashed)
+        email = form.email.data
+        session = q.get_session()
+        user = session.query(db.User).filter_by(email=email).first()
+        if user:
+            raw_token = _create_reset_token(user.id)
+            mail.send_reset_password_link(
+                username=user.username, email=user.email, raw_token=raw_token
+            )
+            return render_template("auth/reset-password-post.html", email=user.email)
+        else:
+            flash(
+                "Sorry, the email address you provided is not associated with any of our acounts."
+            )
+
+    # Override the default message ("The response parameter is missing.")
+    # for better UX.
+    if form.recaptcha.errors:
+        form.recaptcha.errors = ["Please click the reCAPTCHA box."]
+
+    return render_template("auth/reset-password.html", form=form)
+
+
+@bp.route("/reset-password/<username>/<raw_token>", methods=["GET", "POST"])
```

Comment:
We might reduce the risk of exposing tokens in analytics URLs if we send it over as a query string instead.

Reply:
Will leave this for a future revision -- Flask doesn't seem to play well with multiple routes that differ only in their query params.

For now, I updated the site template to disable analytics by default for all atuh-related pages.